viziphant/plotting_unitary_events_analysis.py

```diff
@@ -0,0 +1,358 @@
+            for e_val in params_dict['events'][key]:
+                axes_name.axvline(e_val, ls='-', lw=params_dict['lw'],
+                                  color='r')
+                if axes_name.get_geometry()[2] == 6:
+                    axes_name.text(x=e_val-10*pq.ms, y=-65, s=key, fontsize=9,
+                                   color='r')
+
+    print('plotting Unitary Event Analysis ...')
+
+    print('plotting Spike Events ...')
+    axis1 = plt.subplot(6, 1, 1)
+    axis1.set_title('Spike Events')
+    for n in range(n_neurons):
+        for trial, data_trial in enumerate(data):
+            axis1.plot(data_trial[n].rescale('ms').magnitude,
+                       np.ones_like(data_trial[n].magnitude) * trial +
+                       n * (n_trail + 1) + 1, ls='none', marker='.',
+                       color='k', markersize=0.5)
+        if n < n_neurons - 1:
+            axis1.axhline((trial + 2) * (n + 1), lw=params_dict['lw'],
+                          color='b')
+    axis1.set_xlim(xlim_left, xlim_right)
+    axis1.set_ylim(0, (trial + 2) * (n + 1) + 1)
+    axis1.xaxis.set_major_locator(MaxNLocator(integer=True))
+    axis1.set_yticks(y_ticks_list)
+    axis1.set_yticklabels(y_ticks_labels_list, fontsize=params_dict['fsize'])
+    for n in range(n_neurons):
+        axis1.text(xlim_right + 20, n * (n_trail + 1),
+                   f"Unit {params_dict['unit_real_ids'][n]}")
+    axis1.set_ylabel('Trial', fontsize=params_dict['fsize'])
+
+    print('plotting Spike Rates ...')
+    axis2 = plt.subplot(6, 1, 2, sharex=axis1)
+    axis2.set_title('Spike Rates')
+    # psth = peristimulu time histogram
+    max_val_psth = 0
+    for n in range(n_neurons):
+        axis2.plot(
+            t_winpos + window_size / 2.,
+            joint_surprise_dict['rate_avg'][:, n].rescale('Hz'),
+            label=f"Unit {params_dict['unit_real_ids'][n]}",
+            lw=params_dict['lw'])
+        if max(joint_surprise_dict['rate_avg'][:, n]) > max_val_psth:
+            max_val_psth = max(joint_surprise_dict['rate_avg'][:, n])
+    axis2.set_xlim(xlim_left, xlim_right)
+    max_val_psth = max_val_psth.rescale('Hz').magnitude
+    axis2.set_ylim(0, max_val_psth + max_val_psth/10)
+    axis2.xaxis.set_major_locator(MaxNLocator(integer=True))
+    axis2.set_yticks([0, int(max_val_psth / 2), int(max_val_psth)])
+    mark_epochs(axis2)
+    axis2.legend(bbox_to_anchor=(1, 1), fancybox=True, shadow=True)
+    axis2.set_ylabel('(1/s)', fontsize=params_dict['fsize'])
+
+    print('plotting Coincidence Events ...')
+    axis3 = plt.subplot(6, 1, 3, sharex=axis1)
+    axis3.set_title('Coincidence Events')
+    for n in range(n_neurons):
+        for trial, data_trial in enumerate(data):
+            axis3.plot(
+                data_trial[n].rescale('ms').magnitude,
+                np.ones_like(data_trial[n].magnitude) * trial +
+                n * (n_trail + 1) + 1, ls='none', marker='.', color='k',
+                markersize=0.5)
```

Comment:
can you really read this?
split computation in variables

not only here. go through all lines. I don't want to point at each line, hope you will find them all